src/main/java/io/zipcoder/Classroom.java

```diff
@@ -1,4 +1,143 @@
 package io.zipcoder;
 
+import java.util.Arrays;
+import java.util.HashMap;
+
 public class Classroom {
+
+    protected Student[] students;
+    HashMap<Student, String> gradeBook= new HashMap<>();
+
+    public Classroom() {
+
+        this.students = new Student[30];
+    }
+
+    public Classroom(int maxNumberOfStudents) {
+        this.students = new Student[maxNumberOfStudents];
+    }
+
+    public Classroom(Student[] aStudentArray) {
+        this.students = aStudentArray;
+    }
+
+    public Student[] getStudents() {
+        return this.students;
+
+    }
+
+    public double getClassTotalExamScore() {
+        double classTotalExamScore = 0;
+        for (Student student : this.students) {
+            classTotalExamScore += student.getTotalExamScore();
+        }
+        return classTotalExamScore;
+    }
+
+    public int getClassTotalCountOfExams() {
+        int classTotalCountOfExams = 0;
+        for (Student student : this.students) {
+            classTotalCountOfExams += student.getCountOfExams();
+        }
+        return classTotalCountOfExams;
+    }
+
+    public double getClassAverageExamScore() {
+        double rawAverage = getClassTotalExamScore() / getClassTotalCountOfExams();
```

Comment:
it's kind of sloppy to run thru the loop twice to sum the scores and then to count the number of them

Reply:
Understood, but I wanted to have the ability to sum the scores and/or count the tests separate from the method to calculate the class average.